android/src/main/java/com/urbanairship/reactnative/ReactMessageView.java

```diff
@@ -0,0 +1,159 @@
+/* Copyright Airship and Contributors */
+
+package com.urbanairship.reactnative;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
+import com.urbanairship.Cancelable;
+import com.urbanairship.UAirship;
+import com.urbanairship.richpush.RichPushInbox;
+import com.urbanairship.richpush.RichPushMessage;
+import com.urbanairship.widget.UAWebView;
+import com.urbanairship.widget.UAWebViewClient;
+
+public class ReactMessageView extends UAWebView {
+
+    static final String EVENT_LOAD_STARTED = "onLoadStarted";
+    static final String EVENT_LOAD_FINISHED = "onLoadFinished";
+    static final String EVENT_LOAD_ERROR = "onLoadError";
+    static final String EVENT_CLOSE = "onClose";
+
+    private static final String MESSAGE_ID_KEY = "message";
+    private static final String RETRYABLE_KEY = "retryable";
+    private static final String ERROR_KEY = "error";
+
+    private static final String ERROR_MESSAGE_NOT_AVAILABLE = "MESSAGE_NOT_AVAILABLE";
```

Comment:
Are these values intended to be keys used for constructing human readable errors, or error messages themselves?

Reply:
Sorta both. When an error happens we send an event:

```
"error": "MESSAGE_NOT_AVAILABLE",
"message": "bloop",
"retryable": false
```

Reply:
ok